Kudu.Core/Jobs/BaseJobRunner.cs

```diff
@@ -140,6 +185,10 @@ private void CacheJobBinaries(JobBase job, IJobLogger logger)
                     UpdateAppConfigs(tempJobInstancePath);
 
                     _workingDirectory = tempJobInstancePath;
+
+                    // cache the file map snapshot for next time (to aid in detecting
+                    // file deletions)
+                    _cachedSourceDirectoryFileMap = sourceDirectoryFileMap;
```

Comment:
not sure I understand do you mean line 154? but it relies on the value in `_cachedSourceDirectoryFileMap`.